apps/aehttp/src/aehttpc_aeternity.erl

```diff
@@ -37,59 +38,75 @@ get_generation(Host, Port, Hash) ->
     Path = <<"/v2/generations/hash/", Hash/binary>>,
     get_request(Path, Host, Port, 5000).
 
--spec get_transactions(binary(), integer(), [binary()]) -> {ok, map()} | {error, term()}.
-get_transactions(Host, Port, MBs) ->
+-spec get_transactions(binary(), integer(), [binary()], binary()) -> {ok, list()} | {error, term()}.
+get_transactions(Host, Port, MBs, ParentHCAccountPubKey) ->
     Txs = lists:flatmap(
         fun(MB) ->
-            get_hc_commitments(Host, Port, MB)
+            get_hc_commitments(Host, Port, MB, ParentHCAccountPubKey)
         end, MBs),
     {ok, Txs}.
 
-get_hc_commitments(Host, Port, MB) ->
+get_hc_commitments(Host, Port, MB, ParentHCAccountPubKey) ->
     Path =  <<"/v2/micro-blocks/hash/", MB/binary, "/transactions">>,
     {ok, Res} = get_request(Path, Host, Port, 5000),
     #{<<"transactions">> := Txs} = Res,
     %% TODO - take hc commitment account from some config
     %% Commitments include:
     %%   [{Committer, Committers view of child chain top hash}]
-    Commitments = lists:foldl(
+    ExpectedRecipient = aeser_api_encoder:encode(account_pubkey, ParentHCAccountPubKey),
+    lists:foldl(
             fun(#{<<"tx">> := Tx}, Acc) ->
                     case Tx of
                         #{<<"type">> := <<"SpendTx">>,
-                          <<"recipient_id">> := <<"ak_hc_account">>,
+                          <<"recipient_id">> := ExpectedRecipient,
                           <<"sender_id">> := SenderId,
                           <<"payload">> := Commitment} ->
                             [{SenderId, Commitment} | Acc];
                         _ ->
                             Acc
                     end
-            end, [], Txs),
-    {ok, Commitments}.
+            end, [], Txs).
 
-post_commitment_tx(Host, Port, AccountId, Signature, HCAccountId, CurrentTop) ->
-    %% 1. get the next nonce for our account
-    NoncePath = <<"/v2/accounts/", AccountId/binary, "next-nonce">>,
+post_commitment_tx(Host, Port, AccountId, StakerPrivKey, HCAccountId, CurrentTopHash) ->
+    %% 1. get the next nonce for our account over at the parent chain
+    SenderEnc = aeser_api_encoder:encode(account_pubkey, AccountId),
+    NoncePath = <<"/v2/accounts/", SenderEnc/binary, "/next-nonce">>,
```

Comment:
please use `/v3/` prefix or make the prefix configurable. Our plan is ditching `/v2/`